sympy/vector/coordsysrect.py

```diff
@@ -446,15 +446,18 @@ def lame_coefficients(self):
         return self._lame_coefficients
 
     def transformation_to_parent(self):
-        return self._transformation_lambda(*self.base_scalars())
+        if self._parent is None:
+            raise ValueError("No parent coordinate system, use `transformation_from_parent_function()`")
+        return self._transformation_to_parent_lambda(*self._parent.base_scalars())
```

Comment:
why `self._parent.base_scalars()`? You are transforming the current base scalars to the parent's one.